src/Compilers/CSharp/Test/Emit3/Attributes/AttributeTests_StructLayout.cs

```diff
@@ -722,5 +769,310 @@ public void Bug1075326()
+                }
+
+                partial struct D
+                {
+                    public static int y;
+                }
+
+                [ExtendedLayout(ExtendedLayoutKind.CStruct)] // ok
+                partial struct E
+                {
+                    public const int x = 2;
+                }
+
+                partial struct E
+                {
+                    public int y;
+                }
+                """;
+
+            CreateCompilation(
+                source,
+                references: [MinimalCoreLibBuilder.Create(ExtendedLayoutAttribute).ToMetadataReference()],
+                targetFramework: TargetFramework.Empty)
+            .VerifyDiagnostics(
+                // (5,15): warning CS0282: There is no defined ordering between fields in multiple declarations of partial struct 'C'. To specify an ordering, all instance fields must be in the same declaration.
+                Diagnostic(ErrorCode.WRN_SequentialOnPartialClass, "C").WithArguments("C"));
+        }
+
+        [Fact]
+        public void ExtendedLayout_FieldOffset_Errors()
+        {
+            string source = """
+                using System.Runtime.InteropServices;
+
+                [ExtendedLayout(ExtendedLayoutKind.CStruct)]
+                struct C
+                {
+                    [FieldOffset(4)]
+                    int a;
+                }
+
+                [ExtendedLayout(ExtendedLayoutKind.CUnion)]
+                struct D
+                {
+                    // Even though FieldOffset(0) is equivalent with CUnion, do not support specifying FieldOffset to avoid confusion.
+                    [FieldOffset(0)]
+                    int a;
+                }
+                """;
+
+            CreateCompilation(
+                source,
+                references: [MinimalCoreLibBuilder.Create(ExtendedLayoutAttribute, StructLayoutAttributes).ToMetadataReference()],
+                targetFramework: TargetFramework.Empty)
+            .VerifyDiagnostics(
+                // (6,6): error CS0636: The FieldOffset attribute can only be placed on members of types marked with the StructLayout(LayoutKind.Explicit)
+                //     [FieldOffset(4)]
+                Diagnostic(ErrorCode.ERR_StructOffsetOnBadStruct, "FieldOffset").WithLocation(6, 6),
+                // (13,6): error CS0636: The FieldOffset attribute can only be placed on members of types marked with the StructLayout(LayoutKind.Explicit)
+                //     [FieldOffset(0)]
+                Diagnostic(ErrorCode.ERR_StructOffsetOnBadStruct, "FieldOffset").WithLocation(14, 6)
+                );
+        }
+
+        [Fact]
+        public void ExtendedLayoutAttribute_OtherLayoutKind_Errors()
+        {
+            string source = """
+                using System.Runtime.InteropServices;
+                
+                [StructLayout(LayoutKind.Sequential)]
+                [ExtendedLayout(ExtendedLayoutKind.CStruct)]
```

Comment:
Please have a version with inverted ordering. Please validate what the symbols report for their layouts.